lib/jormungandr/src/Cardano/Wallet/Jormungandr/Network.hs

```diff
@@ -0,0 +1,164 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# OPTIONS_GHC  -fno-warn-redundant-constraints #-}
+
+-- |
+-- Copyright: © 2018-2019 IOHK
+-- License: MIT
+--
+--
+--
+-- This module allows the wallet to retrieve blocks from a known @Jormungandr@
+-- node. This is done by providing a @NetworkLayer@ with some logic building on
+-- top of an underlying @Jormungandr@ HTTP client.
+
+module Cardano.Wallet.Jormungandr.Network
+    ( Jormungandr (..)
+    , mkNetworkLayer
+    , newNetworkLayer
+    , mkJormungandr
+
+    -- * Re-export
+    , BaseUrl (..)
+    , newManager
+    , defaultManagerSettings
+    , Scheme (..)
+    ) where
+
+import Prelude
+
+import Cardano.Wallet.Jormungandr.Api
+    ( BlockId, GetBlock, GetTipId, api )
+import Cardano.Wallet.Network
+    ( ErrNetworkTip (..)
+    , ErrNetworkUnreachable (..)
+    , ErrPostTx (..)
+    , NetworkLayer (..)
+    )
+import Cardano.Wallet.Primitive.Types
+    ( Block, Tx (..), TxWitness (..) )
+import Control.Arrow
+    ( left )
+import Control.Exception
+    ( Exception )
+import Control.Monad.Catch
+    ( throwM )
+import Control.Monad.Trans.Except
+    ( ExceptT (..) )
+import Data.Proxy
+    ( Proxy (..) )
+import Network.HTTP.Client
+    ( Manager, defaultManagerSettings, newManager )
+import Network.HTTP.Types.Status
+    ( status404 )
+import Servant.API
+    ( (:<|>) (..) )
+import Servant.Client
+    ( BaseUrl (..), ClientM, Scheme (..), client, mkClientEnv, runClientM )
+import Servant.Client.Core
+    ( ServantError (..), responseStatusCode )
+import Servant.Links
+    ( Link, safeLink )
+
+-- | Constructs a network layer with the given @Jormungandr@ client.
+mkNetworkLayer :: Monad m => Jormungandr m -> NetworkLayer m
+mkNetworkLayer _httpBridge = NetworkLayer
+    { nextBlocks = undefined
+    , networkTip = undefined
+    , postTx = undefined
+    }
+
+-- | Creates a jormungandr 'NetworkLayer' using the given connection
+-- settings.
+newNetworkLayer
+    :: Int -> IO (NetworkLayer IO)
+newNetworkLayer port = mkNetworkLayer <$> newHttpBridge port
+  where
+    newHttpBridge = undefined
+
+{-------------------------------------------------------------------------------
+                            Jormungandr Client
+-------------------------------------------------------------------------------}
+
+-- | Endpoints of the jormungandr REST API.
+data Jormungandr m = HttpBridge
+    { getBlock
+        :: BlockId -> ExceptT ErrNetworkUnreachable m Block
+    , getTipId
+        :: ExceptT ErrNetworkTip m BlockId
+    , getDecendantIds
+        :: BlockId -> Int -> ExceptT ErrGetDecendants m [BlockId]
+    , postSignedTx
+        :: (Tx, [TxWitness])-> ExceptT ErrPostTx m ()
+    }
```

Comment:
What about not including `getDescendantIds` and `postSignedTx` here since they aren't the subject of this PR ?